src/interpreter_funcs/transform.rs

```diff
@@ -91,25 +100,56 @@ impl Func for FuncTransform {
         let translate = Vector3::from(args[1].unwrap_float3());
         let rotate = args[2].unwrap_float3();
         let scale = Vector3::from(args[3].unwrap_float3());
+        let transform_around_local_center = args[4].unwrap_boolean();
 
-        let translation = Matrix4::new_translation(&translate);
-        let rotation = Rotation::from_euler_angles(
+        let user_rotation = Rotation::from_euler_angles(
             rotate[0].to_radians(),
             rotate[1].to_radians(),
             rotate[2].to_radians(),
         );
-        let scaling = Matrix4::new_nonuniform_scaling(&scale);
+        let user_scaling = Matrix4::new_nonuniform_scaling(&scale);
+        let user_translation = Matrix4::new_translation(&translate);
 
-        let t = translation * Matrix4::from(rotation) * scaling;
+        let value = if transform_around_local_center {
+            // Move to the origin, scale and rotate, then mve back and finally
+            // move according to the user translation.
+            let b_box = BoundingBox::from_meshes(iter::once(mesh));
+            let center = b_box.center();
+            let vector_to_origin = Vector3::zeros() - center.coords;
 
-        let vertices_iter = mesh.vertices().iter().map(|v| t.transform_point(v));
-        let normals_iter = mesh.normals().iter().map(|n| t.transform_vector(n));
+            let translation_to_origin = Matrix4::new_translation(&vector_to_origin);
+            let user_transformation = Matrix4::from(user_rotation) * user_scaling;
+            let translation_from_origin = Matrix4::new_translation(&(-1.0 * vector_to_origin));
+            let final_translation = translation_from_origin * user_translation;
 
-        let value = Mesh::from_faces_with_vertices_and_normals(
-            mesh.faces().iter().copied(),
-            vertices_iter,
-            normals_iter,
-        );
+            let vertices_iter = mesh.vertices().iter().map(|v| {
+                let v1 = translation_to_origin.transform_point(v);
+                let v2 = user_transformation.transform_point(&v1);
+                final_translation.transform_point(&v2)
+            });
+            let normals_iter = mesh.normals().iter().map(|n| {
+                let n1 = translation_to_origin.transform_vector(n);
+                let n2 = user_transformation.transform_vector(&n1);
+                final_translation.transform_vector(&n2)
+            });
+
+            Mesh::from_faces_with_vertices_and_normals(
+                mesh.faces().iter().copied(),
+                vertices_iter,
+                normals_iter,
+            )
```

Comment:
I realize that it would be enough to return only `vertices_iter` and `normals_iter` but `no two closures, even if identical, have the same type`. So it seemed easier to return the mesh itself.